c/anagrams_kata/anagrams_finder.h

```diff
@@ -0,0 +1,14 @@
+#ifndef ANAGRAMS_FINDER_H
+#define ANAGRAMS_FINDER_H
+
+typedef struct anagrams_finder_s
+{
+    char *wordlist;
+    void (*find)(struct anagrams_finder_s *_this, char *word);
```

Comment:
I don't think that this issue has anything to do with performance, so in my opinion it is not sufficient to just refer to "performance as a point of consideration". The problem of reading the whole file at once could lead to memory problems if you run this on very large files. Especially on embedded systems. 

Regarding your question:
I wanted to hide the real implementation of the "find" method. This way i can easily implement different versions of this method and let it just point to desired variant.